Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Use an official Python runtime as a parent image
+FROM python:3.11-slim-buster
+
+# Set the working directory in the container to /app
+WORKDIR /app
+
+# Install git
+RUN apt-get update && \
+    apt-get upgrade -y
+
+# Install Poetry
+RUN pip install poetry
+
+# Copy the current directory contents into the container at /app
+COPY . /app
```

Comment:
**Suggestion:** Use COPY with more granularity to leverage Docker cache layers effectively.
```suggestion
COPY pyproject.toml poetry.lock* /app/
COPY src /app/src
# Continue copying other necessary files and directories individually
```